features/0577-self-encoded-keys/README.md

```diff
@@ -0,0 +1,296 @@
+# Aries RFC 0577: Use self-encoded keys
+
+- Authors: [George Aristy](mailto:george.aristy@securekey.com) (SecureKey)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2020-12-18
+- Status Note: N/A
+- Supersedes: [RFC0360](../0360-use-did-key/README.md)
+- Start Date: 2020-12-09
+- Tags: [feature](/tags.md#feature)
+
+## Summary
+
+A simple, lightweight format for a sender to encode cryptographic keys is introduced that unambiguously conveys the components required for their use: the key's curve, the key's value, and the associated algorithm. Further changes to RFCs with "naked" keys are proposed to disambiguate their purpose.
+
+## Scope
+
+This format is intended to replace the existing usage of `did:key` identifiers in certain Aries RFC protocols until a different or evolved solution is proposed that relies on standards and meets the community's requirements.
+
+## Background
+
+The Hyperledger Aries community standards for encoding of cryptographic keys between a sending a receiving party has gone through two iterations.
+
+### First iteration
+
+The Hyperledger Aries community has historically used only two key types for different purposes in the standard protocols: [Ed25519](https://ed25519.cr.yp.to/ed25519-20110926.pdf) for signatures and [Curve25519](https://cr.yp.to/ecdh/curve25519-20060209.pdf) for key exchange using the [Diffie-Hellman method](https://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.364.5157). These are still the only key types employed by most if not all Hyperledger Aries implementations.
+
+Due to the [birational equivalence](https://eprint.iacr.org/2007/286.pdf) between curves `Ed25519` and `Curve25519`, as well as a general desire to simplify key management, Aries protocol designers and implementers chose a convention whereby the sender transmits the `Ed25519` key only. The receiver assumes this key's purpose as the sender's verification key, and would convert it to the corresponding `Curve25519` key for key exchange with the sender. The `Ed25519` key's bytes were encoded in [Base58](https://en.bitcoinwiki.org/wiki/Base58) format by the sender before transmission.
+
+The above conventions were captured in [RFC160-connection-protocol](https://github.com/hyperledger/aries-rfcs/blob/master/features/0160-connection-protocol/README.md) and [RFC0019-encryption-envelop](https://github.com/hyperledger/aries-rfcs/blob/master/features/0019-encryption-envelope/README.md).
+
+### Second iteration
+
+There was a growing awareness of the drawbacks to the convention implemented in the previous iteration. In particular, there was no support for other types of keys (see [cryptographic agility](https://en.wikipedia.org/wiki/Crypto-agility)), and the sender's purpose for them was not explicit.
```

Comment:
The "no support" line is excessive.  Simply by using multicodec, other keys are supported by definition.  However, I agree with the sentiment that any key that does not support birational equivalence cannot be used in places where both of a pair of keys are needed.

Reply:
@swcurran this paragraph is describing the situation in the first iteration where all naked keys were base58-encoded and were always of type Ed25519, hence "no support for other types of keys".

Regardless, this section is a mere description of past events intended to guide the reader to an understanding of the motivations behind the proposal. It does not form part of the solution's description.

Reply:
Got it.  Thanks.